src/pages/todoList/todoList.js

```diff
@@ -0,0 +1,134 @@
+import { useEffect, useState } from "react";
+import * as T from "./todoList.styles";
+
+export default function TodoListPage() {
+  const [todo, setTodo] = useState("");
+  const [todoList, setTodoList] = useState(() => {
+    // 로컬스토리지에서 할일 불러오기
+    const savedTodo = localStorage.getItem("todoItem");
+    return savedTodo ? JSON.parse(savedTodo) : [];
+  });
+  // 애니메이션 추가,삭제를 위한 인덱스
+  const [addedItemId, setAddedItemId] = useState(null);
+  const [deleteIndexAnimation, setDeleteIndexAnimation] = useState(null);
+
+  // 오늘 날짜
+  const today = new Date().toISOString().slice(0, 10);
+
+  // todoList 변경 시 로컬스토리지 업데이트
+  useEffect(() => {
+    localStorage.setItem("todoItem", JSON.stringify(todoList));
+  }, [todoList]);
+
+  const onChangeTodo = (event) => {
+    setTodo(event.currentTarget.value);
+  };
+
+  const onClickSubmit = (event) => {
+    event.preventDefault();
+    if (!todo.trim()) return;
+
+    const newTodo = { id: Date.now(), text: todo, checked: false };
+    setAddedItemId(newTodo.id);
+    setTodoList([newTodo, ...todoList]);
+
+    // todo 상태 초기화
+    setTodo("");
+  };
+
+  const onClickDelete = (event) => {
+    const indexToDelete = parseInt(event.currentTarget.value, 10);
+
+    setDeleteIndexAnimation(indexToDelete); // 삭제 애니메이션 시작
+    setTimeout(() => {
+      setTodoList(todoList.filter((_, index) => index !== indexToDelete));
+      setDeleteIndexAnimation(null); // 애니메이션 후 상태 업데이트
+    }, 300); // setTimeout의 시간은 애니메이션의 지속 시간과 일치해야 합니다.
+  };
```

Comment:
삭제 애니메이션 멋지네요 ㅎㅎ

다만 `setTimeout`을 사용할 때는 컴포넌트가 언마운트(삭제)되기  전에 타이머를 명확하게 정리해야 메모리 누수를 방지할 수 있대요!!

그래서 `clearTimeout`을 호출해야 하는데, 
useEffect 내부에서 `setTimeout`을 호출하고, 
return문에 `clearTimeout`을 호출해서 
타이머를 정리해주면 
컴포넌트가 사라질 때 메모리 누수를 방지할 수 있다고 합니다!!


```suggestion
  useEffect(() => {
  const timerId = setTimeout(() => {
    setTodoList(todoList.filter((_, index) => index !== indexToDelete));
    setDeleteIndexAnimation(null); 
  }, 300); 

  // 컴포넌트 언마운트되기 전에 clearTimeout 호출하기
  return () => clearTimeout(timerId);
}, [todoList, indexToDelete]);
 // 의존성 배열에 todoList와 indexToDelete추가!!

```

[관련 블로그입니다!!](https://velog.io/@jaewoneee/%EB%A6%AC%EC%95%A1%ED%8A%B8-%EC%BB%B4%ED%8F%AC%EB%84%8C%ED%8A%B8%EC%97%90%EC%84%9C-setTimeout-%EC%82%AC%EC%9A%A9%ED%95%98%EA%B8%B0)

Reply:
좋은 정보 감사합니다! 👍